src/features/farming/shop/Plants.tsx

```diff
@@ -21,17 +21,20 @@ export const Plants: React.FC = () => {
   const [selected, setSelected] = useState<Crop>(CROPS().Sunflower);
   const { setToast } = useContext(ToastContext);
   const [isSellAllModalOpen, showSellAllModal] = React.useState(false);
+  const [isCustomSellModalOpen, showCustomSellModal] = React.useState(false);
   const { gameService } = useContext(Context);
   const [
     {
       context: { state },
     },
   ] = useActor(gameService);
   const [isPriceBoosted, setIsPriceBoosted] = useState(false);
+  const [amountToSell, setAmountToSell] = useState(0);
```

Comment:
are you sure type *number* is okay (or will be okay in future)  ?
Side Note: Inventory uses type *Decimal*.
Also have a look at `L:47`